pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.pinot.core.minion.PinotTaskConfig;
+import org.apache.pinot.spi.annotations.minion.TaskGenerator;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableTaskConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@TaskGenerator
+public class PurgeTaskGenerator implements PinotTaskGenerator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PurgeTaskGenerator.class);
+
+    private static final PurgeTaskGenerator INSTANCE = new PurgeTaskGenerator();
+
+    private ClusterInfoAccessor _clusterInfoAccessor;
+
+    public static Object newInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public void init(ClusterInfoAccessor clusterInfoAccessor) {
+        LOGGER.info("Initializing purge task");
+        _clusterInfoAccessor = clusterInfoAccessor;
+    }
+
+
+    @Override
+    public String getTaskType() {
+        return MinionConstants.PurgeTask.TASK_TYPE;
+    }
+
+
+
+    @Override
+    public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
+        LOGGER.info("Start generating PurgeTask");
+        String taskType = MinionConstants.PurgeTask.TASK_TYPE;
+        List<PinotTaskConfig> pinotTaskConfigs = new ArrayList<>();
+
+        for (TableConfig tableConfig : tableConfigs) {
+
+            String tableName = tableConfig.getTableName();
+            if (tableConfig.getTableType() == TableType.REALTIME) {
+                LOGGER.info("Task type : {}, cannot be run on table of type  {}", taskType, TableType.REALTIME);
+                continue;
+            }
+
+            Map<String, String> taskConfigs;
+            try {
+                TableTaskConfig tableTaskConfig = tableConfig.getTaskConfig();
+                Preconditions.checkNotNull(tableTaskConfig);
+                taskConfigs =
+                        tableTaskConfig.getConfigsForTaskType(MinionConstants.PurgeTask.TASK_TYPE);
+                Preconditions.checkNotNull(taskConfigs, "Task config shouldn't be null for Table: {}", tableName);
+            } catch (Exception e) {
+                continue;
+            }
+            LOGGER.info("Start generating task configs for table: {} for task: {}", tableName, taskType);
+            // Get max number of tasks for this table
+            int tableMaxNumTasks;
+            String tableMaxNumTasksConfig = taskConfigs.get(MinionConstants.TABLE_MAX_NUM_TASKS_KEY);
+            if (tableMaxNumTasksConfig != null) {
+                try {
+                    tableMaxNumTasks = Integer.parseInt(tableMaxNumTasksConfig);
+                } catch (Exception e) {
+                    tableMaxNumTasks = Integer.MAX_VALUE;
+                }
+            } else {
+                tableMaxNumTasks = Integer.MAX_VALUE;
+                LOGGER.warn("MaxNumTasks have been incorrectly set for table : {}, and task {}", tableName, taskType);
```

Comment:
This message seems incorrect though, it's just that the config didn't set up. Maybe this line should be moved after Line 101?